### PR TITLE
[SCFToCalyx] Add SCFToCalyx boilerplate [1/13]

### PR DESCRIPTION
**Pull request by mortbopet**
This is the initial commit for the SCFToCalyx pass. The commit sets up the boilerplate required for adding a new conversion pass.

### CODE REVIEW

File: include/circt/Conversion/Passes.td [hunk 1]
```diff
@@ -15,6 +15,21 @@
 
 include "mlir/Pass/PassBase.td"
 
+//===----------------------------------------------------------------------===//
+// SCFToCalyx
+//===----------------------------------------------------------------------===//
+
+def SCFToCalyx : Pass<"lower-scf-to-calyx", "mlir::ModuleOp"> {
+  let summary = "Lower SCF/Standard to Calyx";
+  let description = [{This pass lowers SCF / standard to calyx.}];
```
Comment (cgyurgyik) [inline include/circt/Conversion/Passes.td:24]: Nit: capitalized Calyx.

File: lib/Conversion/SCFToCalyx/SCFToCalyx.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+using namespace llvm;
+using namespace mlir;
+
+namespace circt {
+
+//===----------------------------------------------------------------------===//
+// Pass driver
+//===----------------------------------------------------------------------===//
+class SCFToCalysPass : public SCFToCalyxBase<SCFToCalysPass> {
```
Comment (cgyurgyik) [inline lib/Conversion/SCFToCalyx/SCFToCalyx.cpp:27]: SCFToCalys -> SCFToCalyx

**Review by cgyurgyik — APPROVED** [review]
LGTM!

**Review by mikeurbach — APPROVED** [review]
Thanks!

**Pull request merged**